Solver factory cleanup

thomasspriggs opened the pull request:
A couple of small tidy ups I wanted to do after reading the solver factory to do other work. This clean up is raised as a separate PR as it doesn't depend on the other work or vice versa
<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

thomasspriggs committed:
Add `final` to `solver_factory` / `solvert`

This aids readability because the reader will know that no classes
derive from these classes.

peterschrammel commented on src/goto-checker/solver_factory.cpp:
```diff
@@ -411,7 +411,6 @@ void solver_factoryt::no_incremental_check()
 {
   const bool all_properties = options.get_bool_option("all-properties");
   const bool cover = options.is_set("cover");
-  const bool incremental_check = options.is_set("incremental-check");
```
This should be `incremental-loop`.

thomasspriggs replied:
Ok. I have removed this deletion from this PR and pushed a fix with a regression test instead.

thomasspriggs committed:
Fix handling of `--incremental-loop` with non-inremental solver

This check was previously broken because the name of the option in the
check did not match the name used in the command line parsing.
